TABTestKit/Classes/Elements/DatePicker.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  DatePicker.swift
+//  Pods-TABTestKit_Example
+//
+//  Created by Annie Fung on 06/12/2019.
+//
+
+import XCTest
+
+/// Represents a datepicker.
+/// To interact with the wheels you must first
+/// ask the picker for one of it's wheels, by index.
+
+public struct DatePicker: Element {
+	
+	public let id: String?
+	public let index: Int
+	public let parent: Element
+	public let type: XCUIElement.ElementType = .datePicker
+	
+	/// Returns the number of wheels the picker contains.
+	public var numberOfWheels: Int { return underlyingXCUIElement.pickerWheels.count }
+	
+	public init(id: String, index: Int = 0, parent: Element = App.shared) {
+		self.id = id
+		self.index = index
+		self.parent = parent
+	}
+	
+	public func wheel(_ index: Int) -> Wheel {
+		return Wheel(index: index, picker: self)
+	}
+}
+
+public extension DatePicker {
+	
+	struct Wheel: Element, ValueRepresentable, Adjustable {
+		
+		public let id: String? = nil
+		public let index: Int
+		public let parent: Element
+		public let type: XCUIElement.ElementType = .pickerWheel
+		public var value: String { return underlyingXCUIElement.value as? String ?? "" }
+		
+		public init(index: Int, picker: DatePicker) {
+			self.index = index
+			self.parent = picker
+		}
+		
+		public func adjust(to newValue: String) {
+			underlyingXCUIElement.adjust(toPickerWheelValue: newValue)
```

Comment:
Should we `guard` here that the value isn't already what you're trying to set it to? We do that in `Picker.swift`

Reply:
> Should we `guard` here that the value isn't already what you're trying to set it to? We do that in `Picker.swift`

I remove the `guard` on purpose. The way how the `DatePicker` works on our project app is that when the user taps on it, today's date will show on the `DatePicker` automatically. If the date set in your test matches with today's day or month, the test will fail (for example today's date is 4/12/2019 and the test is trying to adjust the `DatePicker` to 5/12/2009). This failure is always going to happen and I think having that check in the `guard` is too rigid for a `DatePicker`.

Reply:
Sorry I don't full understand the problem. If your tests are trying to set it to something that it's already set to then that's an issue with the tests. Your tests should know what the expected date to change from and to is?

>for example today's date is 4/12/2019 and the test is trying to adjust the DatePicker to 5/12/2009

Why would that make the test fail if there's a guard? The guard is checking that you're not trying to set it to the thing it's already set to, which means that changing from the 4th to the 5th wouldn't fail?

Reply:
In this example, today's date is 4/12/2019 and the test is trying to adjust the DatePicker to 5/12/2009, will fail because the month is set to December in both cases

Reply:
So just change the day wheel and not the month wheel?

Reply:
yes I remembered that's what Ben and I found out when implementing the datePicker on the project app, the value you set in the test (either, day, month or year) cannot match with today's date, the test will fail if either value matches.

Reply:
Hmm can you show me an example soon?